lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h

```diff
@@ -0,0 +1,1001 @@
+  amf_app_timer_t m5_paging_response_timer;
+
+  /* send_ue_purge_request: If true AMF shall send - Purge Req to
+   * delete contexts at HSS
+   */
+  bool send_ue_purge_request;
+  bool hss_initiated_detach;
+  bool location_info_confirmed_in_hss;
+
+  /* S6a- update location request guard timer */
+  amf_app_timer_t m5_ulr_response_timer;
+  uint8_t registration_type;
+  lai_t lai;
+
+  /* granted_service_t: informs the granted service to UE */
+  // TODO required during dual connectivity (DC)
+  // granted_service_t m5_granted_service;
+  /*  paging_proceeding_flag (PPF) shall set to true, when UE moves to
+   * connected. Indicates that paging procedure can be prooceeded, Is set to
+   * false, due to "Inactivity of UE including lack of periodic TAU"
+   */
+  bool ppf;
+
+#define SUBSCRIPTION_UNKNOWN false
+#define SUBSCRIPTION_KNOWN true
+  bool subscription_known;
+  ambr_t used_ambr;
+  subscriber_status_t subscriber_status;
+  network_access_mode_t network_access_mode;
+  bool path_switch_req;
+};
+
+/** @class ue_m5gmm_context_s
+ *  @brief Useful parameters to know in AMF application layer. They are set
+ * according to 3GPP TS.23.518 #6.1.6.2.25
+ */
+class ue_mm_context {
+ public:
+  ue_mm_context() {}
+  ~ue_mm_context() {}
+
+  /* msisdn: The basic MSISDN of the UE. The presence is dictated by its storage
+   *         in the UDM, set by N8 UPDATE LOCATION ANSWER
+   */
+  std::string imsi;
+  bool supi_UnauthInd;
+  std::string pei;
+  uint64_t udmGroupId;   // NfGroupId
+  uint64_t ausfGroupId;  // NfGroupId;
+  std::string routingIndicator;
+  std::string drxParameter;
+  std::string subRfsp;
+  uint32_t usedRfsp;  // RfspIndex type;
+  ambr_t subUeAmbr;
+  bool smsSupport;
+  std::string smsfId;  // NfInstanceId type
+  std::string
+      seafData;  // SeafData which will come while AUSF communication for AUTN.
+  std::string pcfId;                   // NfInstanceId
+  std::string pcfAmPolicyUri;          // Uri
+  std::string pcfUePolicyUri;          // Uri
+  std::string hpcfId;                  // NfInstanceId
+  std::string restrictedRatList;       // array(RatType)
+  std::string forbiddenAreaList;       // array(Area)
+  std::string serviceAreaRestriction;  // ServiceAreaRestriction
+  std::string restrictedCnList;        // array(CoreNetworkType)
+  std::string eventSubscriptionList;   // array(AmfEventSubscription)
+  std::string mmContextList;           // array(MmContext)
+  std::string sessionContextList;      // array(PduSessionContext)
+  std::string traceData;               // TraceData
+
+  /* SCTP assoc id */
+  sctp_assoc_id_t sctp_assoc_id_key;
+
+  /* gNB UE NGAP ID,  Unique identity the UE within gNodeB */
+  gnb_ue_ngap_id_t gnb_ue_ngap_id : 24;
+  gnb_ngap_id_key_t gnb_ngap_id_key;
+
+  /* MME UE S1AP ID, Unique identity the UE within MME */
+  amf_ue_ngap_id_t amf_ue_ngap_id;
+};
+
+class amf_app_ue_context : public amf_ue_context_t, public ue_m5gmm_context_s {
```

Comment:
inheritance is quite needless here. The inherited classes simple structs, define them as private class variables in `class amf_app_ue_context`.

Reply:
fixed